### PR TITLE
Update to bevy 0.10

### LINKED ISSUE #32: Update to Bevy 0.10
Seeing as Bevy 0.10 is about to be released, I thought it's time to put this in the room 🙂

### PR DESCRIPTION
**Pull request by Elabajaba**
There's a couple things I'm not sure on, but otherwise it's basically done.

Closes #32 

edit: It actually works now, just the two open questions remaining.

### CODE REVIEW

File: src/render_dim.rs [hunk 1]
```diff
@@ -47,17 +47,18 @@ pub mod r3d {
...
-                shader_defs.push("DEPTH_TEST_ENABLED".to_string());
+                shader_defs.push("DEPTH_TEST_ENABLED".into());
             }
 
             let vertex_buffer_layout = layout.get_layout(&[
                 Mesh::ATTRIBUTE_POSITION.at_shader_location(0),
                 Mesh::ATTRIBUTE_COLOR.at_shader_location(1),
             ])?;
             let (label, blend, depth_write_enabled);
-            if key.contains(MeshPipelineKey::TRANSPARENT_MAIN_PASS) {
```
Comment (Elabajaba) [inline src/render_dim.rs:60]: I'm not sure if I have to specialize this for each `BLEND_*` mode now. (ALPHA, MULTIPLY, OPAQUE, and PREMULTIPLIED_ALPHA)

File: src/lib.rs [hunk 2]
```diff
@@ -135,8 +137,9 @@ impl Plugin for DebugLinesPlugin {
             })
             .init_resource::<dim::DebugLinePipeline>()
             .init_resource::<SpecializedMeshPipelines<dim::DebugLinePipeline>>()
-            .add_system_to_stage(RenderStage::Extract, extract)
-            .add_system_to_stage(RenderStage::Queue, dim::queue);
+            // TODO: Is ExtractCommands the right set?
```
Comment (Elabajaba) [inline src/lib.rs:140]: I think ExtractCommands is Extract, but I'm not 100% sure.

File: src/lib.rs [hunk 3]
```diff
@@ -126,7 +126,9 @@ impl Plugin for DebugLinesPlugin {
         app.init_resource::<DebugShapes>();
 
         app.add_startup_system(setup)
-            .add_system_to_stage(CoreStage::PostUpdate, update.label("draw_lines"));
+            .add_system(update.in_base_set(CoreSet::PostUpdate));
```
Comment (Elabajaba) [inline src/lib.rs:129]: Not sure if we can stack sets, and if I have to make draw_lines not a string?

File: src/lib.rs (hunk 2 above)
Comment (Elabajaba) [reply src/lib.rs]: This was wrong, I needed to add it to the `ExtractSchedule`.

File: src/lib.rs (hunk 3 above)
Comment (Elabajaba) [reply src/lib.rs]: I nested the sets and it seems to work?

File: src/lib.rs [hunk 4]
```diff
@@ -68,6 +68,12 @@ pub(crate) struct DebugLinesConfig {
     depth_test: bool,
 }
 
+
+#[derive(SystemSet, Debug, Hash, PartialEq, Eq, Clone)]
+pub enum DebugLinesSet {
+    DrawLines,
```
Comment (Elabajaba) [inline src/lib.rs:74]: I'm not sure if this enum should be public or not (or if we even need to nest this into it's own set).

Comment (Toqozz) [reply src/lib.rs:74]: I think we just leave it public for the sake of simplicity.  If people need to access it then they can.  If they don't, that's fine.

File: src/render_dim.rs (hunk 1 above)
Comment (Toqozz) [reply src/render_dim.rs:60]: I'm not entirely sure.  It should just be for the pass that the lines shader uses right, which should be just `BLEND_ALPHA`.

The bevy pull request looks like they just do `BLEND_ALPHA`, so should be good.  Not sure if they have toggle-able depth testing.
```rust
            fragment: Some(FragmentState {
                shader: self.shader.clone_weak(),
                shader_defs: vec![],
                entry_point: "fragment".into(),
                targets: vec![Some(ColorTargetState {
                    format: TextureFormat::bevy_default(),
                    blend: Some(BlendState::ALPHA_BLENDING),
                    write_mask: ColorWrites::ALL,
                })],
            }),
```

**Review by Toqozz — COMMENTED** [review]
Awesome, thanks for the PR!  Hopefully we get https://github.com/bevyengine/bevy/pull/6529 soon (0.10 maybe??)

I'll merge and test this properly when 0.10 gets released.

**Comment by Elabajaba** [comment]
Updated the examples, but the 3D ones are broken.

```
2023-03-06T21:14:13.120381Z ERROR bevy_render::render_resource::pipeline_cache: failed to process shader:
error: expected expression, found '#'
   ┌─ wgsl:59:41
   │
59 │     cascades: array<DirectionalCascade, #{MAX_CASCADES_PER_LIGHT}>,
   │                                         ^ expected expression

```

**Comment by Elabajaba** [comment]
Everything seems to be working now.

**Comment by Elabajaba** [comment]
Do you squash on merge? Or would you like me to squash this?

**Comment by Toqozz** [comment]
I'll squash it.  Will merge shortly.

**Pull request merged**